### DISCUSSION TITLE
`LinkNeighborLoader` weighted sampling

### DISCUSSION THREAD

**Discussion by denadai2** [pyg-team/pytorch_geometric#7876] [Q&A] [answered]
Dear community,

I'd like to implement `LinkNeighborLoader` with a weighted sampling, giving a probability to each edge or node. Is there a trick to do it with the current pyg/pyg-lib/torch-scatter? If no, could you help me implementing it? I am not sure what's the best way to start e.g in pyg-lib

thx

**Comment by rusty1s** [comment, marked as answer]
We are currently working on this with - should land pretty soon. Progress will be tracked in https://github.com/pyg-team/pytorch_geometric/issues/7331

**Comment by denadai2** [comment]
That's a GREAT news! Eager to ear more about it!
